### PR TITLE
[CLOUD-2432] Update EAP CD templates to use edge TLS

### PR DESCRIPTION
**Pull request by luck3y**
https://issues.jboss.org/browse/CLOUD-2432
Signed-off-by: Ken Wills <kwills@redhat.com>

Note this also adds (Passthrough TLS) to the template names where appropriate. tx-recovery was skipped, as thats really more of an example only.

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request title is properly formatted: `[CLOUD-XYA] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

**Review by luck3y — COMMENTED** [review]
Opps, ignore this, I'll update...

File: templates/eap-cd-amq-persistent-s2i.json [hunk 1]
```diff
@@ -6,10 +6,10 @@
             "iconClass": "icon-eap",
             "tags": "eap,javaee,java,jboss",
             "version": "14.0",
-            "openshift.io/display-name": "JBoss EAP CD + AMQ 7 (with https)",
+            "openshift.io/display-name": "JBoss EAP CD + AMQ 7 with https (Passthrough TLS)",
```
Comment (rcernich) [inline templates/eap-cd-amq-persistent-s2i.json]: I think you can remove `with https` as it seems redundant.  It also shortens the name.

**Review by rcernich — DISMISSED** [review]
Looks good to me.

File: dev-overrides.yaml [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
+#version: "14.0"
+#from: "jboss-eap-7/eap-cd:14.0"
+
 modules:
       repositories:
           - name: cct_module
             git:
-                  url: https://github.com/jboss-openshift/cct_module.git
-                  ref: master
+                  url: https://github.com/luck3y/cct_module.git
```
Comment (Crumby) [inline dev-overrides.yaml]: Updating references to personal repository doesn't seem correct.

File: dev-overrides.yaml [hunk 3]
```diff
@@ -1,17 +1,20 @@
...
-                  ref: master
+                  url: https://github.com/luck3y/jboss-eap-modules.git
+                  ref: CLOUD-2758
+
 osbs:
       repository:
             name: containers/jboss-eap-7
-            branch: jb-eap-7.2-cd-dev-openshift-rhel-7
-
+            branch: jb-eap-7.1-openshift-dev-rhel-7
```
Comment (Crumby) [inline dev-overrides.yaml]: Do we want to change target branch to 7.1 for CD releases?

**Review by Crumby — CHANGES_REQUESTED** [review]

File: templates/eap-cd-amq-persistent-s2i.json (hunk 1 above)
Comment (bstansberry) [reply templates/eap-cd-amq-persistent-s2i.json]: +1. "with passthrough TLS" is good.  No capital P, no parens.

File: templates/eap-cd-amq-s2i.json [hunk 4]
```diff
@@ -6,10 +6,10 @@
             "iconClass": "icon-eap",
             "tags": "eap,javaee,java,jboss,hidden",
             "version": "14.0",
-            "openshift.io/display-name": "JBoss EAP CD + AMQ 7 (with https)",
+            "openshift.io/display-name": "JBoss EAP CD + AMQ 7 with https (Passthrough TLS)",
```
Comment (bstansberry) [inline templates/eap-cd-amq-s2i.json]: "with passthrough TLS"

File: templates/eap-cd-https-s2i.json [hunk 5]
```diff
@@ -6,10 +6,10 @@
             "iconClass": "icon-eap",
             "tags": "eap,javaee,java,jboss,hidden",
             "version": "14.0",
-            "openshift.io/display-name": "JBoss EAP CD (with https)",
+            "openshift.io/display-name": "JBoss EAP CD with https (Passthrough TLS)",
```
Comment (bstansberry) [inline templates/eap-cd-https-s2i.json]: "with passthrough TLS"

File: templates/eap-cd-mongodb-persistent-s2i.json [hunk 6]
```diff
@@ -6,10 +6,10 @@
             "iconClass": "icon-eap",
             "tags": "eap,javaee,java,jboss,hidden",
             "version": "14.0",
-            "openshift.io/display-name": "JBoss EAP CD + MongoDB (with https)",
+            "openshift.io/display-name": "JBoss EAP CD + MongoDB with https (Passthrough TLS)",
```
Comment (bstansberry) [inline templates/eap-cd-mongodb-persistent-s2i.json]: "with passthrough TLS"

File: templates/eap-cd-mysql-persistent-s2i.json [hunk 7]
```diff
@@ -6,10 +6,10 @@
             "iconClass": "icon-eap",
             "tags": "eap,javaee,java,jboss",
             "version": "14.0",
-            "openshift.io/display-name": "JBoss EAP CD + MySQL (with https)",
+            "openshift.io/display-name": "JBoss EAP CD + MySQL with https (Passthrough TLS)",
```
Comment (bstansberry) [inline templates/eap-cd-mysql-persistent-s2i.json]: "with passthrough TLS"

File: templates/eap-cd-sso-s2i.json [hunk 8]
```diff
@@ -6,10 +6,10 @@
             "iconClass": "icon-eap",
             "tags": "eap,javaee,java,jboss,hidden",
             "version": "14.0",
-            "openshift.io/display-name": "JBoss EAP CD + Single Sign-On (with https)",
+            "openshift.io/display-name": "JBoss EAP CD + Single Sign-On with https (Passthrough TLS)",
```
Comment (bstansberry) [inline templates/eap-cd-sso-s2i.json]: "with passthrough TLS"

File: templates/eap-cd-basic-s2i.json [hunk 9]
```diff
@@ -6,10 +6,10 @@
             "iconClass": "icon-eap",
             "tags": "eap,javaee,java,jboss",
             "version": "14.0",
-            "openshift.io/display-name": "JBoss EAP CD (no https)",
+            "openshift.io/display-name": "JBoss EAP CD",
             "openshift.io/provider-display-name": "Red Hat, Inc.",
             "description": "An example JBoss Enterprise Application Platform continuous delivery application. For more information about using this template, see https://github.com/jboss-container-images/jboss-eap-7-openshift-image/blob/eap-cd/README.adoc",
-            "template.openshift.io/long-description": "This template defines resources needed to develop a JBoss Enterprise Application Platform continuous delivery based application, including a build configuration, application deployment configuration and insecure communication using http.",
+            "template.openshift.io/long-description": "This template defines resources needed to develop a JBoss Enterprise Application Platform continuous delivery based application, including a build configuration, application deployment configuration and secure communication using https (Edge TLS).",
```
Comment (bstansberry) [inline templates/eap-cd-basic-s2i.json]: s/https (Edge TLS)/edge TLS/g

**Review by bstansberry — CHANGES_REQUESTED** [review]

**Comment by luck3y** [comment]
Changes made, should be g2g now.

**Review by bstansberry — APPROVED** [review]

**Pull request merged**